pennylane/optimize/gradient_descent.py

```diff
@@ -47,7 +47,7 @@ def update_stepsize(self, stepsize):
         """
         self._stepsize = stepsize
 
-    def step(self, objective_fn, x, grad_fn=None):
+    def step_and_cost(self, objective_fn, x, grad_fn=None):
         """Update x with one step of the optimizer.
```

Comment:
```suggestion
        """Update x with one step of the optimizer and return the corresponding objective function value.
```

Reply:
I just realised that it is kind of confusing that the method name is `step_and_cost`, but the argument name is `objective_fn`.